runtime/connection_cache.go

```diff
@@ -110,7 +110,7 @@ func (r *Runtime) openAndMigrate(ctx context.Context, cfg cachedConnectionConfig
 		}
 	}
 
-	handle, err := drivers.Open(cfg.driver, cfg.instanceID, cfg.config, activityClient, logger)
+	handle, err := drivers.Open(cfg.driver, cfg.instanceID, cfg.config, r.storage.WithPrefix(cfg.instanceID), activityClient, logger)
```

Comment:
I think we should include the connector name here (e.g. `WithPrefix(cfg.instanceID, cfg.name)`) to correctly namespace the connectors. This would mirror the old behavior which also includes the `name` in the prefix.
```
	res.setPreset("data_dir", r.DataDir(instanceID, name), true)
```

It would need to track the `name` in `cachedConnectionConfig`. This change is already present in this open PR, you can lift it from there and then I'll fix the merge conflicts: https://github.com/rilldata/rill/pull/6134

Reply:
Yeah I planned to do this as a follow up. I wasn't too sure if `name` should be part of cache key or not.
Can port changes from your PR :)